### PR TITLE
Make the retrying snap tasks switching

### PR DESCRIPTION
**Pull request by pinges**
## PR description
Convert snap tasks from retrying (same peer) to retrying switching (different peer per retry).

This PR was compared to release 24.6.0 with the following result:
To retrieve 1.96 billion nodes to sync the world state it took 5 nodes  
    - 24.6.0:        2356 minutes (on average 471.2 minutes)
    - this PR:       2164 minutes (on average 432.8 minutes)

or on average the nodes from this PR were 38.4 minutes faster.

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/RetryingGetBytecodeFromPeerTask.java [hunk 1]
```diff
@@ -17,19 +17,21 @@
...
 import java.util.concurrent.CompletableFuture;
 
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
 
-public class RetryingGetBytecodeFromPeerTask extends AbstractRetryingPeerTask<Map<Bytes32, Bytes>> {
+public class RetryingGetBytecodeFromPeerTask
+    extends AbstractRetryingSwitchingPeerTask<Map<Bytes32, Bytes>> {
+
+  public static final int MAX_RETRIES = 4;
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/RetryingGetBytecodeFromPeerTask.java:34]: is this MAX_RETRIES per peer?

**Review by macfarla — COMMENTED** [review]
makes sense! maybe a changelog entry?

**Review by jframe — APPROVED** [review]
LGTM - nice improvement

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/RetryingGetBytecodeFromPeerTask.java (hunk 1 above)
Comment (pinges) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/RetryingGetBytecodeFromPeerTask.java:34]: That is retries with other peers.

**Pull request merged**